flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSource.java

```diff
@@ -100,6 +100,9 @@ public SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> restoreEnu
                     // Only full compaction results will appear on the max level.
                     plan = scan.withLevel(table.options().numLevels() - 1).plan();
                 } else {
+                    if (!isContinuous) {
+                        scan.withReadCompacted(table.options().readCompacted());
```

Comment:
You have modified connector code, it is better to add cases in `ContinuousFileStoreITCase` and `BatchFileStoreITCase` too.